velox/common/caching/CachedFactory.h

```diff
@@ -65,7 +75,9 @@ class CachedFactory {
   // The function returns a pair. The boolean in the pair indicates whether a
   // cache hit or miss. The Value is the generator output for the key if cache
   // miss, or Value in the cache if cache hit.
-  std::pair<bool, Value> generate(const Key& key);
+  std::pair<bool, Value> generate(
+      const Key& key,
+      const Metadata& metadata = Metadata());
```

Comment:
Let's pass in the rvalue here, we want to put something with unique ownership in the future.